packages/spear-cli/src/utils/dom.ts

```diff
@@ -124,7 +125,190 @@ export async function generateAliasPagesFromPagesList(
+
+      if (loopElement && hasTagLoop) {
+        // Routing Hook
+        for (const plugin of settings.plugins) {
+          if (plugin.routing) {
+            try {
+              const pages = await plugin.routing(page);
+              if (pages) {
+                replacePagesList.push(...pages);
+              }
+            } catch (e) {
+              console.warn(` plugin process failed. [${plugin.pluginName}]`)
+            }
+          }
+        }
+        // combination of pagination and tag
+        const tagFieldName = loopElement.getAttribute("cms-tag-loop");
+        const contentType  = loopElement.getAttribute("cms-content-type");
+        const loopId       = loopElement.getAttribute("cms-loop-id");
+        if (!tagFieldName) throw new Error("You should specify the cms-tag-loop");
+        if (!contentType)  throw new Error("You should specify the cms-content-type");
+
+        const apiOption = generateAPIOptionMap(loopElement as Element);
+        apiOption.set('limit', settings.maxPaginationCount);
+        removeCMSAttributes(loopElement as Element);
+        if (settings.debugMode) {
+          loopElement.setAttribute("data-spear-content-type", `{%= ${contentType}_#content_type %}`);
+          loopElement.setAttribute("data-spear-content", `{%= ${contentType}_#alias %}`);
+        }
+        const generatedContents = await jsGenerator.generateEachContentFromList(
+          loopElement.outerHTML,
+          contentType,
+          apiOption,
+          tagFieldName,
+          settings.debugMode
+        );
+        const targetLoopElementHTMLTemplate = loopElement.outerHTML;
+        const targetPageHTMLTemplate = page.node.innerHTML;
+        const tagElements: Map<string, Array<{element: Element, count: number, currentPage: number}>> = new Map();
+        generatedContents.forEach((c, i) => {
+          const tags = c.tag;
+          tags.forEach(tag => {
+            if (!tagElements.has(tag)) {
+              const newElement = parse("") as Element;
+              tagElements.set(tag, [{ element: newElement, count: 0, currentPage: 1 }]);
+            }
+            const obj = tagElements.get(tag);
+            const element = obj[obj.length - 1].element;
+            const count = obj[obj.length - 1].count;
+            const currentPage = obj[obj.length - 1].currentPage;
+            const generatedHTML = c.generatedHtml;
+            element.innerHTML += generatedHTML;
+            obj[obj.length - 1] = { element: element, count: count + 1, currentPage: currentPage };
+            if ((count + 1 ) >=  paginationSize) {
+              const newElement = parse("") as Element;
+              newElement.innerHTML = "";
+              obj.push({ element: newElement, count: 0, currentPage: currentPage + 1 });
+            }
+            tagElements.set(tag, obj);
+          });
+        });
+        tagElements.forEach((v, k) => {
+          const tag = k;
```

Comment:
配列でもないので、直接 `tag` って指定して良さそうです 🤔 

```suggestion
        tagElements.forEach((v, tag) => {
```